lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/VectorIntBitUnpackers.java

```diff
@@ -0,0 +1,1478 @@
+/*
```

Comment:
Did you write this by hand or is it generated class?

Reply:
All hand crafted, there are enough special cases that writing code to generate this code would have been quite complex as well